anomalib/utils/metrics/__init__.py

```diff
@@ -68,3 +51,125 @@ def metric_collection_from_names(metric_names: List[str], prefix: Optional[str])
         else:
             warnings.warn(f"No metric with name {metric_name} found in Anomalib metrics or TorchMetrics.")
     return metrics
+
+
+def _validate_metrics_dict(metrics: Dict[str, Dict[str, Any]]) -> None:
+    """Check the assumptions about metrics config dict.
+
+    - Keys are metric names
+    - Values are dictionaries.
+    - Internal dictionaries:
+        - have key "class_path" and its value is of type str
+        - have key init_args" and its value is of type dict).
+
+    """
+    assert all(
+        isinstance(metric, str) for metric in metrics.keys()
+    ), f"All keys (metric names) must be strings, found {sorted(metrics.keys())}"
+    assert all(
+        isinstance(metric, (dict, DictConfig)) for metric in metrics.values()
+    ), f"All values must be dictionaries, found {list(metrics.values())}"
+    assert all("class_path" in metric and isinstance(metric["class_path"], str) for metric in metrics.values()), (
+        "All internal dictionaries must have a 'class_path' key whose value is of type str, "
+        f"found {list(metrics.values())}"
+    )
+    assert all(
+        "init_args" in metric and isinstance(metric["init_args"], (dict, DictConfig)) for metric in metrics.values()
+    ), (
+        "All internal dictionaries must have a 'init_args' key whose value is of type dict, "
+        f"found {list(metrics.values())}"
+    )
+
+
+def _get_class_from_path(class_path: str) -> Any:
+    """Get a class from a module assuming the string format is `package.subpackage.module.ClassName`."""
+    module_name, class_name = class_path.rsplit(".", 1)
+    module = importlib.import_module(module_name)
+    assert hasattr(module, class_name), f"Class {class_name} not found in module {module_name}"
+    cls = getattr(module, class_name)
+    return cls
+
+
+def metric_collection_from_dicts(metrics: Dict[str, Dict[str, Any]], prefix: Optional[str]) -> AnomalibMetricCollection:
+    """Create a metric collection from a dict of "metric name" -> "metric specifications".
+
+    Example:
+
+    metrics = {
```

Comment:
Thanks for adding the example. I think you need to indent the section contents for sphinx to correctly format the contents.

Reply:
Sorry, I don't get it. Isn't it already indented?
Could you propose the change as a suggestion?

Reply:
Maybe it is and github is formatting it incorrectly but I mean metrics and the other lines should not be on the same column as Example. 
It is currently 
```python
    Example:

    metrics = {
        "PixelWiseF1Score": {
            "class_path": "torchmetrics.F1Score",
            "init_args": {},
        },
```
and I mean

```python
    Example:

        metrics = {
            "PixelWiseF1Score": {
                "class_path": "torchmetrics.F1Score",
                "init_args": {},
            },
```
Not sure if this example makes it clear.

Reply:
done @ashwinvaidya17